editor/src/app/app.component.ts

```diff
@@ -88,9 +91,14 @@ export class AppComponent implements OnInit {
   }
 
   hideOverlay() {
-    this.store.dispatch(AppHideOverlay);
-    this.router.navigate(['/']);
+    this.inputFocus$.pipe(take(1)).subscribe((isInputFocused) => {
```

Comment:
you could add a filter here instead of if statement:
```ts
this.inputFocus$.pipe(
    take(1),
    filter(isInputFocused => isInputFocused)
).subscribe(() => {
    this.store.dispatch(AppHideOverlay);
    this.router.navigate(['/']); 
})
```

Reply:
👍